app/models/heartbeat.rb

```diff
@@ -24,4 +24,15 @@ def last_action
   def action_count
     SubjectAction.where(workflow_id: @workflow.id).count + UserAction.where(workflow_id: @workflow.id).count
   end
+
+  def subject_count
```

Comment:
maybe there's a raw sql query to count these that wouldn't require three separate DB calls?

Reply:
That's a good idea, I don't drop back to SQL as often as it would be useful

Reply:
Or just build an AR relation that includes everything you want and call `count` on it to make a single call instead of summing multiples.

Reply:
i'm reworking this whole bit for another pull request because of the slow page load time, so this advice is proving quite helpful